lib/vnet/app_resolver.go

```diff
@@ -160,20 +185,67 @@ func newTCPAppHandler(
 	leafClusterName string,
 	app types.Application,
 ) (*tcpAppHandler, error) {
+	dialOpts, err := appProvider.GetDialOptions(ctx, profileName)
+	if err != nil {
+		return nil, trace.Wrap(err, "getting dial options for profile %q", profileName)
+	}
+
+	appCertIssuer := &appCertIssuer{
+		appProvider:     appProvider,
+		profileName:     profileName,
+		leafClusterName: leafClusterName,
+		app:             app,
+	}
+	middleware := client.NewCertChecker(appCertIssuer, nil /*clock*/)
```

Comment:
Thoughts on passing through a clock here? 😶 From experience I know that not doing that from the get go makes adding tests harder. OTOH if we don't expect this part to be tested because of VNet root requirements then it doesn't matter.

Reply:
added the clock and some tests that exercise cert refreshes